### PR TITLE
Fix some common redundancies in C#

### PR DESCRIPTION
**Pull request by dampir**
I removed  some common C# redundancies  in whole solution. I used Resharper for it. It makes code more clean and robust.

### CODE REVIEW

**Comment by olegtk** [comment]
Awesome, thanks! :+1:

File: src/FixMixedTabs/Impl/MarginFactory.cs [hunk 1]
```diff
@@ -18,16 +18,16 @@ namespace FixMixedTabs
     internal sealed class MarginFactory : IWpfTextViewMarginProvider
     {
         [Import]
-        private ITextDocumentFactoryService _TextDocumentFactoryService = null;
-        [Import]
+        private ITextDocumentFactoryService _TextDocumentFactoryService;
```
Comment (olegtk) [inline src/FixMixedTabs/Impl/MarginFactory.cs]: I'm getting error CS0649: Field 'MarginFactory._TextDocumentFactoryService' is never assigned to, and will always have its default value null.
That's a warning, but this project is setup to treat warnings as errors.

Is it building OK for you? Unfortunately appveyor is still not functioning.

Comment (dampir) [inline src/FixMixedTabs/Impl/MarginFactory.cs]: No, project cannot be built on my PC too. Anyway, I return these null assignments back.

**Comment by dampir** [comment]
@olegtk I've fixed this error, but anyway there are build error on CI

**Comment by olegtk** [comment]
Thanks! CI is not yet working, ignore it for now.

**Pull request merged**